src/attached_probe.cpp

```diff
@@ -803,8 +819,66 @@ void AttachedProbe::load_prog()
   cache_progfd();
 }
 
+#ifdef HAVE_LIBBPF_KPROBE_MULTI
+static inline uint64_t ptr_to_u64(const void *ptr)
+{
+  return (uint64_t)(unsigned long)ptr;
+}
+
+void AttachedProbe::attach_multi_kprobe(void)
+{
+  DECLARE_LIBBPF_OPTS(bpf_link_create_opts, opts);
+
+  const char **syms = new const char *[probe_.funcs.size()];
```

Comment:
How about directly using `probe_.funcs.data()` instead of allocating a new array? I think `.data()` satisfies the C API's needs here (contiguous memory)

Reply:
Oh, it needs an array of C strings. Still, how about using a `std::vector<char *>` instead of raw pointer? So we don't need to worry about freeing memory here (atm there's a leak)

Reply:
ah right, will check on that, thanks